commit ca6790e0208a5513098a9657420a8261af077ded
Author: QbProg <tholag@gmail.com>
Date:   Thu Mar 17 13:12:18 2011 +0100

    Replaced guards for MSVC specific pragmas from WNT to _MSC_VER
    
    These are mainly #pragma warnings or MSVCRT specific things.
    Left #pragma comment (lib, "") because other compilers seem to support these too.

From 68c342332a2f6348765af35510ee397db05d5a82 Mon Sep 17 00:00:00 2001
From: Muellni <mmuellen@gmx.de>
Date: Tue, 19 Jul 2011 12:32:38 +0200
Subject: [PATCH 06/26] proper MSVC compiler check to fix compilation errors
 for other windows compilers

---
 src/OSD/OSD_MAllocHook.cxx |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

Index: oce/src/BOPTools/BOPTools_PaveSet.cxx
===================================================================
--- oce.orig/src/BOPTools/BOPTools_PaveSet.cxx
+++ oce/src/BOPTools/BOPTools_PaveSet.cxx
@@ -26,7 +26,7 @@
 #include <BOPTools_QuickSortPave.hxx>
 #include <BOPTools_ComparePave.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( disable : 4101 )
 #endif
 
@@ -89,6 +89,6 @@
     }
   }
 }
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( default : 4101 )
 #endif
Index: oce/src/BinDrivers/BinDrivers.cxx
===================================================================
--- oce.orig/src/BinDrivers/BinDrivers.cxx
+++ oce/src/BinDrivers/BinDrivers.cxx
@@ -33,7 +33,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/BinLDrivers/BinLDrivers.cxx
===================================================================
--- oce.orig/src/BinLDrivers/BinLDrivers.cxx
+++ oce/src/BinLDrivers/BinLDrivers.cxx
@@ -30,7 +30,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/BinTObjDrivers/BinTObjDrivers.cxx
===================================================================
--- oce.orig/src/BinTObjDrivers/BinTObjDrivers.cxx
+++ oce/src/BinTObjDrivers/BinTObjDrivers.cxx
@@ -32,7 +32,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/BinXCAFDrivers/BinXCAFDrivers.cxx
===================================================================
--- oce.orig/src/BinXCAFDrivers/BinXCAFDrivers.cxx
+++ oce/src/BinXCAFDrivers/BinXCAFDrivers.cxx
@@ -28,7 +28,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/BooleanOperations/BooleanOperations_ShapesDataStructure.cxx
===================================================================
--- oce.orig/src/BooleanOperations/BooleanOperations_ShapesDataStructure.cxx
+++ oce/src/BooleanOperations/BooleanOperations_ShapesDataStructure.cxx
@@ -59,7 +59,7 @@
 #include <BOPTColStd_Dump.hxx>
 
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( disable : 4291 )
 #endif
 
Index: oce/src/DPrsStd/DPrsStd.cxx
===================================================================
--- oce.orig/src/DPrsStd/DPrsStd.cxx
+++ oce/src/DPrsStd/DPrsStd.cxx
@@ -38,7 +38,7 @@
 #include <OSD_File.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 //static Handle(AppStdL_Application) stdApp;
Index: oce/src/FWOSDriver/FWOSDriver.cxx
===================================================================
--- oce.orig/src/FWOSDriver/FWOSDriver.cxx
+++ oce/src/FWOSDriver/FWOSDriver.cxx
@@ -26,7 +26,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/IntTools/IntTools.cxx
===================================================================
--- oce.orig/src/IntTools/IntTools.cxx
+++ oce/src/IntTools/IntTools.cxx
@@ -38,7 +38,7 @@
 #include <TColStd_ListIteratorOfListOfReal.hxx>
 #include <gce_ErrorType.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( disable : 4101 )
 #endif
 //=======================================================================
@@ -378,6 +378,6 @@
   return 1;
 }
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( default : 4101 )
 #endif
Index: oce/src/IntTools/IntTools_EdgeFace.cxx
===================================================================
--- oce.orig/src/IntTools/IntTools_EdgeFace.cxx
+++ oce/src/IntTools/IntTools_EdgeFace.cxx
@@ -69,7 +69,7 @@
 #include <GeomAdaptor_HSurface.hxx>
 #include <IntCurveSurface_IntersectionPoint.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( disable : 4101 )
 #endif
 
@@ -1577,6 +1577,6 @@ Standard_Integer AdaptiveDiscret (const
 }
 
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning ( default : 4101 )
 #endif
Index: oce/src/OSD/OSD_Environment.cxx
===================================================================
--- oce.orig/src/OSD/OSD_Environment.cxx
+++ oce/src/OSD/OSD_Environment.cxx
@@ -243,7 +243,9 @@ Standard_Integer OSD_Environment::Error(
 
 #include <windows.h>
 
+#ifdef _MSC_VER
 #pragma warning( disable : 4700 )
+#endif
 
 static void __fastcall _set_error ( OSD_Error&, DWORD );
 
Index: oce/src/OSD/OSD_MAllocHook.cxx
===================================================================
--- oce.orig/src/OSD/OSD_MAllocHook.cxx
+++ oce/src/OSD/OSD_MAllocHook.cxx
@@ -85,7 +85,7 @@ OSD_MAllocHook::CollectBySize* OSD_MAllo
 // Platform-dependent methods
 //!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
-#ifdef WNT
+#ifdef _MSC_VER
 #include <crtdbg.h>
 
 static long getRequestNum(void* pvData, long lRequest, size_t& theSize)
@@ -164,7 +164,7 @@ void OSD_MAllocHook::SetCallback(Callbac
     _CrtSetAllocHook(MyAllocHook);
 }
 
-#else // ! WNT
+#else // ! _MSC_VER
 
 // Not yet implemented for non-WNT platform
 
@@ -173,7 +173,7 @@ void OSD_MAllocHook::SetCallback(Callbac
   MypCurrentCallback = theCB;
 }
 
-#endif // WNT
+#endif // _MSC_VER
 
 //!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 // LogFileHandler handler methods
Index: oce/src/OSD/OSD_Process.cxx
===================================================================
--- oce.orig/src/OSD/OSD_Process.cxx
+++ oce/src/OSD/OSD_Process.cxx
@@ -236,8 +236,9 @@ Standard_Integer OSD_Process::Error()con
 #include <OSD_WNT_1.hxx>
 #include <LMCONS.H> /// pour UNLEN  ( see MSDN about GetUserName() )
 
-
+#ifdef _MSC_VER
 #pragma warning( disable : 4700 )
+#endif
 
 void _osd_wnt_set_error ( OSD_Error&, OSD_WhoAmI, ... );
 
Index: oce/src/OpenGl/OpenGl_GraphicDriver.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_GraphicDriver.cxx
+++ oce/src/OpenGl/OpenGl_GraphicDriver.cxx
@@ -42,7 +42,7 @@ namespace
 // classe OSD_SharedLibrary dans la methode SetGraphicDriver de la
 // classe Graphic3d_GraphicDevice
 extern "C" {
-#ifdef WNT /* disable MS VC++ warning on C-style function returning C++ object */
+#ifdef _MSC_VER /* disable MS VC++ warning on C-style function returning C++ object */
   #pragma warning(push)
   #pragma warning(disable:4190)
 #endif
@@ -51,7 +51,7 @@ extern "C" {
     Handle(OpenGl_GraphicDriver) aOpenDriver = new OpenGl_GraphicDriver (AShrName);
     return aOpenDriver;
   }
-#ifdef WNT
+#ifdef _MSC_VER
   #pragma warning(pop)
 #endif
 }
Index: oce/src/PS/PS_Driver.cxx
===================================================================
--- oce.orig/src/PS/PS_Driver.cxx
+++ oce/src/PS/PS_Driver.cxx
@@ -123,7 +123,7 @@ static const char* MONTHS [] = {
 #include <Image_Image.hxx>
 #include <stdio.h>
 static Handle(Image_Image) myImage;
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning (disable : 4244 4018 4101)
 #endif
 
Index: oce/src/PlotMgt/PlotMgt_TextManager.cxx
===================================================================
--- oce.orig/src/PlotMgt/PlotMgt_TextManager.cxx
+++ oce/src/PlotMgt/PlotMgt_TextManager.cxx
@@ -21,7 +21,7 @@
 #include <PlotMgt_TextManager.ixx>
 #include <Aspect_Units.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning (disable : 4244)
 #endif
 
Index: oce/src/Poly/Poly_CoherentNode.cxx
===================================================================
--- oce.orig/src/Poly/Poly_CoherentNode.cxx
+++ oce/src/Poly/Poly_CoherentNode.cxx
@@ -22,7 +22,7 @@
 #include <Poly_CoherentNode.hxx>
 #include <Poly_CoherentTriangle.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(disable:4996)
 #endif
 
Index: oce/src/StdDrivers/StdDrivers.cxx
===================================================================
--- oce.orig/src/StdDrivers/StdDrivers.cxx
+++ oce/src/StdDrivers/StdDrivers.cxx
@@ -31,7 +31,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/StdLDrivers/StdLDrivers.cxx
===================================================================
--- oce.orig/src/StdLDrivers/StdLDrivers.cxx
+++ oce/src/StdLDrivers/StdLDrivers.cxx
@@ -30,7 +30,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/TObjDRAW/TObjDRAW.cxx
===================================================================
--- oce.orig/src/TObjDRAW/TObjDRAW.cxx
+++ oce/src/TObjDRAW/TObjDRAW.cxx
@@ -46,7 +46,7 @@
 #include <Draw_PluginMacro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/TestTopOpe/TestTopOpe.cxx
===================================================================
--- oce.orig/src/TestTopOpe/TestTopOpe.cxx
+++ oce/src/TestTopOpe/TestTopOpe.cxx
@@ -29,7 +29,7 @@
 #include <Draw_Interpretor.hxx>
 #include <Draw_PluginMacro.hxx>
 #include <BOPTest.hxx>
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/TopExp/TopExp_Explorer.cxx
===================================================================
--- oce.orig/src/TopExp/TopExp_Explorer.cxx
+++ oce/src/TopExp/TopExp_Explorer.cxx
@@ -143,7 +143,7 @@ const TopoDS_Shape&  TopExp_Explorer::Cu
 //purpose  : 
 //=======================================================================
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(push)
 #pragma warning(disable: 4291) // to avoid warning when using new(buffer) syntax
 #endif
@@ -225,7 +225,7 @@ void  TopExp_Explorer::Next()
   hasMore = Standard_False;
 }
 
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(pop)
 #endif
 
Index: oce/src/ViewerTest/ViewerTest.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest.cxx
+++ oce/src/ViewerTest/ViewerTest.cxx
@@ -68,7 +68,7 @@
 #include <Viewer2dTest.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning(4:4190)
 #pragma warning (disable:4996)
Index: oce/src/ViewerTest/ViewerTest_ObjectCommands.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest_ObjectCommands.cxx
+++ oce/src/ViewerTest/ViewerTest_ObjectCommands.cxx
@@ -142,7 +142,7 @@
 #include <strings.h>
 #endif
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/ViewerTest/ViewerTest_ViewerCommands.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest_ViewerCommands.cxx
+++ oce/src/ViewerTest/ViewerTest_ViewerCommands.cxx
@@ -74,8 +74,10 @@
 #include <WNT_WClass.hxx>
 #include <WNT_Window.hxx>
 
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
+#endif
 
 #endif
 
Index: oce/src/VrmlData/VrmlData_Geometry.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Geometry.cxx
+++ oce/src/VrmlData/VrmlData_Geometry.cxx
@@ -38,7 +38,7 @@
 #include <NCollection_Vector.hxx>
 #include <Standard_ErrorHandler.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_Group.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Group.cxx
+++ oce/src/VrmlData/VrmlData_Group.cxx
@@ -29,7 +29,7 @@
 #include <Precision.hxx>
 #include <gp_Ax1.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_IndexedFaceSet.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_IndexedFaceSet.cxx
+++ oce/src/VrmlData/VrmlData_IndexedFaceSet.cxx
@@ -35,7 +35,7 @@
 #include <Poly.hxx>
 #include <TShort_HArray1OfShortReal.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_IndexedLineSet.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_IndexedLineSet.cxx
+++ oce/src/VrmlData/VrmlData_IndexedLineSet.cxx
@@ -30,7 +30,7 @@
 #include <TopoDS_Edge.hxx>
 #include <TopoDS_Wire.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_Material.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Material.cxx
+++ oce/src/VrmlData/VrmlData_Material.cxx
@@ -24,7 +24,7 @@
 #include <VrmlData_Scene.hxx>
 #include <gp_XYZ.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_Node.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Node.cxx
+++ oce/src/VrmlData/VrmlData_Node.cxx
@@ -29,7 +29,7 @@
 #include <gp_XY.hxx>
 #include <gp_XYZ.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_Scene.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Scene.cxx
+++ oce/src/VrmlData/VrmlData_Scene.cxx
@@ -48,7 +48,7 @@
 #include <BRep_Builder.hxx>
 #include <Precision.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/VrmlData/VrmlData_WorldInfo.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_WorldInfo.cxx
+++ oce/src/VrmlData/VrmlData_WorldInfo.cxx
@@ -23,7 +23,7 @@
 #include <VrmlData_Scene.hxx>
 #include <VrmlData_InBuffer.hxx>
 
-#ifdef WNT
+#ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning (disable:4996)
 #endif
Index: oce/src/XCAFDrivers/XCAFDrivers.cxx
===================================================================
--- oce.orig/src/XCAFDrivers/XCAFDrivers.cxx
+++ oce/src/XCAFDrivers/XCAFDrivers.cxx
@@ -35,7 +35,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/XDEDRAW/XDEDRAW.cxx
===================================================================
--- oce.orig/src/XDEDRAW/XDEDRAW.cxx
+++ oce/src/XDEDRAW/XDEDRAW.cxx
@@ -94,7 +94,7 @@
 
 #define ZVIEW_SIZE 1000000.0
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/XSDRAWSTLVRML/XSDRAWSTLVRML.cxx
===================================================================
--- oce.orig/src/XSDRAWSTLVRML/XSDRAWSTLVRML.cxx
+++ oce/src/XSDRAWSTLVRML/XSDRAWSTLVRML.cxx
@@ -78,7 +78,7 @@
 #include <Bnd_Box.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/XmlDrivers/XmlDrivers.cxx
===================================================================
--- oce.orig/src/XmlDrivers/XmlDrivers.cxx
+++ oce/src/XmlDrivers/XmlDrivers.cxx
@@ -39,7 +39,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/XmlLDrivers/XmlLDrivers.cxx
===================================================================
--- oce.orig/src/XmlLDrivers/XmlLDrivers.cxx
+++ oce/src/XmlLDrivers/XmlLDrivers.cxx
@@ -34,7 +34,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/XmlTObjDrivers/XmlTObjDrivers.cxx
===================================================================
--- oce.orig/src/XmlTObjDrivers/XmlTObjDrivers.cxx
+++ oce/src/XmlTObjDrivers/XmlTObjDrivers.cxx
@@ -33,7 +33,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
Index: oce/src/XmlXCAFDrivers/XmlXCAFDrivers.cxx
===================================================================
--- oce.orig/src/XmlXCAFDrivers/XmlXCAFDrivers.cxx
+++ oce/src/XmlXCAFDrivers/XmlXCAFDrivers.cxx
@@ -28,7 +28,7 @@
 #include <Plugin_Macro.hxx>
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
-#ifdef WNT
+#ifdef _MSC_VER
 #pragma warning(4:4190)
 #endif
 
